src/sidebar/components/SidebarPanel.tsx

```diff
@@ -56,28 +59,42 @@ export default function SidebarPanel({
     }
   }, [panelIsActive, onActiveChanged]);
 
-  const closePanel = useCallback(() => {
-    store.toggleSidebarPanel(panelName, false);
-  }, [store, panelName]);
+  const onTransitionEnd = useCallback(
+    (direction: 'in' | 'out') => {
+      setShowPanelContent(direction === 'in');
+
+      if (direction !== 'in') {
+        return;
+      }
+
+      const focusEl = initialFocus?.current as HTMLElement & {
+        disabled?: boolean;
+      };
+
+      if (focusEl && !focusEl.disabled) {
+        focusEl.focus();
+      }
+    },
+    [initialFocus],
+  );
+
+  const closePanel = useCallback(
+    () => store.toggleSidebarPanel(panelName, false),
+    [store, panelName],
+  );
 
   return (
-    <>
-      {panelIsActive && (
-        <Dialog
-          initialFocus={initialFocus}
-          restoreFocus
-          ref={panelElement}
-          classes="mb-4"
-          title={title}
-          icon={icon}
-          onClose={closePanel}
-          transitionComponent={Slider}
-          variant={variant}
-          scrollable={false}
-        >
-          {children}
-        </Dialog>
-      )}
-    </>
+    <CloseableContext.Provider value={{ onClose: closePanel }}>
+      <Slider
+        direction={panelIsActive ? 'in' : 'out'}
+        onTransitionEnd={onTransitionEnd}
+      >
+        {mountContent && (
+          <section aria-label={description} className="mb-4">
```

Comment:
I used `div[role="region"]` in my first iteration, but according to MDN, using `section` has the same effect and is preferred: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/region_role#prefer_html